### PR TITLE
[Defender (RiskIQ) - EASM] Api version 2024-03-01-preview 

### LINKED ISSUE #28855: [RiskIQ - EASM - Defender EASM] API Review
New API Review meeting has been requested.

**Service Name**: RiskIQ - EASM - Defender EASM
**Review Created By**: Adam An
**Review Date**: 05/02/2024 04:00 PM PT
**Release Plan**: [](https://web.powerapps.com/apps/821ab569-ae60-420d-8264-d7b5d5ef734c?release-plan-id=)
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/28322
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=6ace1f6e-7703-ef11-a1fd-000d3a34b75c)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by juyoung98**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/riskiq/data-plane/Microsoft.Easm/preview/2023-03-01-preview/easm.json
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/28855
fix https://github.com/Azure/azure-rest-api-specs/issues/28855

### CODE REVIEW

**Comment by chunyu3** [comment]
.Net CI pipeline failed because the model name broke the .NET SDK Analyzer rule AZC0030 (see following). You may need to customize the model's name for csharp or ask approval from arch board and then suppress it.
```
error AZC0030: Model name 'ObservationResponse' ends with 'Response'. Suggest to rename it to 'ObservationResult' or any other appropriate name. [/mnt/vss/_work/1/s/azure-sdk-for-net/sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj::TargetFramework=netstandard2.0]
```

**Comment by MaryGao** [comment]
I checked this [pr](https://github.com/azure-sdk/azure-sdk-for-js/pull/9226) locally and the code could be generated correctly but we have an issue when generating apiView ([tracked](https://github.com/Azure/autorest.typescript/issues/2404) here). Will take a look when back from holiday.

File: specification/riskiq/Easm/main.tsp [hunk 1]
```diff
@@ -40,7 +40,11 @@ using TypeSpec.Versioning;
 namespace Easm;
 @doc("The EASM Defender service version.")
 enum Versions {
-  @doc("Version 2023-03-01-preview")
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "Documentation will be added next preview version"
```
Comment (weidongxu-microsoft) [inline specification/riskiq/Easm/main.tsp:43]: I don't see the reason why you replace a `doc` with a `suppress`?

File: specification/riskiq/Easm/models.tsp [hunk 2]
```diff
@@ -270,6 +323,39 @@ model AssetUpdateData {
   transfers?: AssetUpdateTransfers;
 }
 
+@doc("A request body used to export an asset.")
+@Versioning.added(Versions.v2024_03_01_preview)
+model AssetsExportRequestBody {
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "Documentation will be added next preview version"
```
Comment (weidongxu-microsoft) [inline specification/riskiq/Easm/models.tsp:329]: I don't think this "Documentation will be added next preview version" is a valid approach.

**Review by weidongxu-microsoft — COMMENTED** [review]
There seems to be major changes to the new api-version.

Make sure you had scheduled an API review meeting.

**Comment by juyoung98** [comment]
> There seems to be major changes to the new api-version.
> 
> Make sure you had scheduled an API review meeting.

There are no breaking changes so this PR wasn't labeled that we need a review. We are only adding new endpoints, not making breaking changes. Do we still need a review? If so, why?

File: specification/riskiq/Easm/main.tsp (hunk 1 above)
Comment (juyoung98) [reply specification/riskiq/Easm/main.tsp:43]: this was a mistake. We will add it back. Thanks

**Comment by weidongxu-microsoft** [comment]
> > There seems to be major changes to the new api-version.
> > Make sure you had scheduled an API review meeting.
> 
> There are no breaking changes so this PR wasn't labeled that we need a review. We are only adding new endpoints, not making breaking changes. Do we still need a review? If so, why?

https://eng.ms/docs/products/azure-developer-experience/design/api-review

File: specification/riskiq/Easm/models.tsp [hunk 3]
```diff
@@ -270,6 +323,42 @@ model AssetUpdateData {
...
+  @doc("The type of asset")
+  kind?: GlobalAssetType;
+
+  @doc("expected format to be: yyyy-MM-dd")
+  date?: string;
+}
+
+@doc("A request body used to retrieve a delta summary.")
+@Versioning.added(Versions.v2024_03_01_preview)
+model DeltaSummaryRequestBody {
```
Comment (johanste) [inline specification/riskiq/Easm/models.tsp:354]: Let's remove the "body" suffix. Is either value required? What happens if you provide neither?

**Review by JeffreyRichter — APPROVED** [review]

File: specification/riskiq/Easm/models.tsp [hunk 4]
```diff
@@ -1671,9 +2034,121 @@ model MarkParameter {
...
+  })
+  @doc("The properties to include in the response.")
+  responseIncludes?: string[];
+}
+
+@doc("The prameter of responseIncludes")
+@Versioning.added(Versions.v2024_03_01_preview)
+model RecentOnlyParameter {
+  @query("recentOnly")
+  @doc("If it's recent only.")
```
Comment (weidongxu-microsoft) [inline specification/riskiq/Easm/models.tsp:2059]: How "recent" is recent?

File: specification/riskiq/Easm/models.tsp [hunk 5]
```diff
@@ -1671,9 +2034,121 @@ model MarkParameter {
...
+@Versioning.added(Versions.v2024_03_01_preview)
+model DiscoGroupSummaryResult {
+  @doc("The system generated unique id for the resource.")
+  id: string;
+
+  @doc("The caller provided unique name for the resource.")
+  name: string;
+
+  @doc("The name that can be used for display purposes.")
+  displayName: string;
```
Comment (weidongxu-microsoft) [inline specification/riskiq/Easm/models.tsp:2153]: Are we sure these properties are always returned?

Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp:2153]: yes by default this display name would be the same as the property name, if it's not specified.

File: specification/riskiq/Easm/models.tsp [hunk 6]
```diff
@@ -270,6 +323,42 @@ model AssetUpdateData {
...
+  deltaDetailType: DeltaDetailType;
+
+  @doc("The number of days prior to retrieve deltas for.")
+  daysPrior?: int32;
+
+  @doc("The type of asset")
+  kind: GlobalAssetType;
+
+  @doc("expected format to be: yyyy-MM-dd")
+  date?: string;
```
[the remark this answers is not in the document]
Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp:349]: no it doesn't

File: specification/riskiq/Easm/models.tsp (hunk 4 above)
Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp:2059]: it varies. this recent is a internal term we use, definition is below:

<html>
<body>
<span><span class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak" dir="ltr"><h2 style="margin-bottom: 16px; margin-top: 16px;"><strong>RECENCY</strong></h2><p style="margin-bottom: 16px; margin-top: 8px;">Currently GI API and UI search both default to searching against "recent" data. There is a special dropdown in the UI and a search history API that will search against historical data.</p><p style="margin-bottom: 16px; margin-top: 8px;">We define recency based on the datasource from which the various asset fields come from and whether the last seen on those fields makes the cutoff.

File: specification/riskiq/Easm/models.tsp (hunk 3 above)
Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp]: removed and also marked value as required

File: specification/riskiq/Easm/models.tsp (hunk 6 above)
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply specification/riskiq/Easm/models.tsp:349]: So, what time zone would this date be?

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/riskiq/Easm/models.tsp [hunk 7]
```diff
@@ -1539,10 +1828,84 @@ union SslCertAssetValidationType {
...
+}
+
+@doc("The result response for the observation")
+@Versioning.added(Versions.v2024_03_01_preview)
+model ObservationResult {
+  @doc("The name of the observation.")
+  name: string;
+
+  @doc("The list of applicable types.")
+  type: ObservationType[];
```
[the remark this answers is not in the document]
Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp]: change to types

File: specification/riskiq/Easm/models.tsp (hunk 6 above)
[the remark this answers is not in the document]
Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp:349]: it's utc time

File: specification/riskiq/Easm/models.tsp [hunk 8]
```diff
@@ -270,6 +323,42 @@ model AssetUpdateData {
...
+}
+
+@doc("A request body used to retrieve a list of deltas.")
+@Versioning.added(Versions.v2024_03_01_preview)
+model DeltaDetailsRequest {
+  @doc("The type of delta detail to retrieve.")
+  deltaDetailType: DeltaDetailType;
+
+  @doc("The number of days prior to retrieve deltas for.")
+  daysPrior?: int32;
```
[the remark this answers is not in the document]
Comment (adamgitazure) [reply specification/riskiq/Easm/models.tsp]: changed to priorDays

**Pull request merged**